src/main/java/core/rating/RatingCalculationCache2.java

```diff
@@ -0,0 +1,26 @@
+package core.rating;
+
+import java.util.HashMap;
+
+abstract class RatingCalculationCache2<T1, T2> {
```

Comment:
What is the purpose of the `RationCalculationCache` classes?  I wonder if we could not use generic methods instead of introducing new classes for each?